test/runtests.jl

```diff
@@ -40,6 +40,13 @@ end
 @assert isa(hess_op(model, [0.]), LinearOperator)
 @assert isa(jac_op(model, [0.]), LinearOperator)
 
+# MathProgNLSModel with user-defined functions
+model = mgh35()
+@assert model.meta.name == "mgh35"
+
+model = mgh07()
+@assert model.meta.name == "mgh07"
```

Comment:
Could we also test that user functions are accessible (e.g., by evaluating the residuals)?